### PR TITLE
PIN-4658 BKE - purpose process - Add route reject purpose

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**
It needs [PIN-4657](https://github.com/pagopa/interop-be-purpose-management/pull/213) to compile

[PIN-4657]: https://pagopa.atlassian.net/browse/PIN-4657?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala [hunk 1]
```diff
@@ -498,6 +498,42 @@ final case class PurposeApiServiceImpl(
...
+              PurposeManagementDependency.ChangedBy.PRODUCER,
+              dateTimeSupplier.get()
+            )
+          )
+        )
+        _              <- authorizationManagementService.updateStateOnClients(
+          purposeId = purposeUUID,
+          versionId = versionUUID,
+          state = AuthorizationManagementDependency.ClientComponentState.INACTIVE
+        )
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: This part is not necessary.
@galales, are you agree?

Comment (galales) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: I agree, only a `WaitingForApproval` version can be rejected, and in that state it cannot be in `ACTIVE` state on the client, so disabling it is not necessary

File: src/main/scala/it/pagopa/interop/purposeprocess/service/impl/PurposeManagementServiceImpl.scala [hunk 2]
```diff
@@ -131,6 +131,17 @@ final case class PurposeManagementServiceImpl(invoker: PurposeManagementInvoker,
...
   }
 
+  def rejectPurposeVersion(purposeId: UUID, versionId: UUID, payload: RejectPurposeVersionPayload)(implicit
+    contexts: Seq[(String, String)]
+  ): Future[Unit] = withHeaders { (bearerToken, correlationId) =>
+    val request =
+      api.rejectPurposeVersion(xCorrelationId = correlationId, purposeId, versionId, payload)(BearerToken(bearerToken))
+    invoker.invoke(
+      request,
+      s"Rejecting Version $versionId of Purpose $purposeId with reason ${payload.rejectionReason}"
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/purposeprocess/service/impl/PurposeManagementServiceImpl.scala]: As rule of thumb we avoid logging free input text
```suggestion
      s"Rejecting Version $versionId of Purpose $purposeId"
```

File: src/main/resources/interface-specification.yml [hunk 3]
```diff
@@ -741,6 +789,14 @@ components:
       schema:
         type: string
   schemas:
+    RejectPurposeVersionPayload:
+      type: object
+      description: models the reject payload for this purpose version.
+      properties:
+        rejectionReason:
+          type: string
```
Comment (galales) [inline src/main/resources/interface-specification.yml:797]: [field constraints](https://github.com/pagopa/interop-be-backend-for-frontend/pull/269/files#diff-17a3d713b849787a0ed57552f2b367a47eedf4cef81a6ba025d0060ac107d213R10295-R10296) should go in the process

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**